src/ScriptEngine/ScriptEngine.cs

```diff
@@ -76,22 +76,33 @@ void LoadDLL(string path, GameObject obj)
                     dll.Write(ms);
                     var ass = Assembly.Load(ms.ToArray());
 
-                    foreach(Type type in ass.GetTypes())
+                    try
                     {
-                        if(typeof(BaseUnityPlugin).IsAssignableFrom(type))
+                        foreach (Type type in ass.GetTypes())
                         {
-                            var metadata = MetadataHelper.GetMetadata(type);
-                            if(metadata != null)
+                            if (typeof(BaseUnityPlugin).IsAssignableFrom(type))
                             {
-                                var typeDefinition = dll.MainModule.Types.First(x => x.FullName == type.FullName);
-                                var typeInfo = Chainloader.ToPluginInfo(typeDefinition);
-                                Chainloader.PluginInfos[metadata.GUID] = typeInfo;
-
-                                Logger.Log(LogLevel.Info, $"Reloading {metadata.GUID}");
-                                StartCoroutine(DelayAction(() => obj.AddComponent(type))); 
+                                var metadata = MetadataHelper.GetMetadata(type);
+                                if (metadata != null)
+                                {
+                                    var typeDefinition = dll.MainModule.Types.First(x => x.FullName == type.FullName);
+                                    var typeInfo = Chainloader.ToPluginInfo(typeDefinition);
+                                    Chainloader.PluginInfos[metadata.GUID] = typeInfo;
+
+                                    Logger.Log(LogLevel.Info, $"Reloading {metadata.GUID}");
+                                    StartCoroutine(DelayAction(() => obj.AddComponent(type)));
+                                }
                             }
                         }
                     }
+                    catch (ReflectionTypeLoadException e)
+                    {
+                        StringBuilder strTypes = new StringBuilder();
+                        foreach (var t in e.Types) { strTypes.Append(t + "\r\n"); }
+                        StringBuilder strExceptions = new StringBuilder();
+                        foreach (var l in e.LoaderExceptions) { strExceptions.Append(l + "\r\n"); }
```

Comment:
The curly brackets are not needed for single-line `foreach` blocks. In addition, put the single line on a newline.